### ISSUE TITLE
Bad performance while searching string

### ISSUE THREAD

**Issue by Chandler-Sun** [balderdashy/sails-mongo#195]
https://github.com/balderdashy/sails-mongo/blob/master/lib/query/index.js#L338

It should not replace the string to regular expression by default, unless it contains a "%". Because it will cause full collection scan in MongoDB if the search keyword is a regular expression and cannot get any benefits from index.

**Comment by sailsbot** [comment]
Thanks for posting, @Chandler-Sun.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
